src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,112 @@
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long productId) {
+        String query = "SELECT * FROM products WHERE product_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, productId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Product product = getProductFromSet(resultSet);
+                return Optional.of(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessException("Can't find product with id " + productId, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products WHERE deleted = FALSE";
+        List<Product> allProducts = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery(query);
+            while (resultSet.next()) {
+                Product product = getProductFromSet(resultSet);
+                allProducts.add(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessException("Can't execute query!", e);
+        }
+        return allProducts;
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE products "
+                + "SET productName = ?, price = ? WHERE product_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, product.getName());
+            statement.setDouble(2, product.getPrice());
+            statement.setLong(3, product.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessException("Can't update product", e);
+        }
+        return product;
+    }
+
+    @Override
+    public boolean delete(Long productId) {
+        String query = "UPDATE products "
+                + "SET deleted = TRUE WHERE product_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, productId);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessException("Can't execute update product " + productId, e);
+        }
+    }
+
+    private Product getProductFromSet(ResultSet resultSet) {
+        try {
+            String productName = resultSet.getString("productName");
```

Comment:
```suggestion
            String productName = resultSet.getString("product_name");
```
use same naming style

Reply:
if name should differ i would change String var name, cuz  `productName` its my column name in DB

Reply:
done